src/web_connection_base.cpp

```diff
@@ -113,7 +113,11 @@ namespace libtorrent
 
 	void web_connection_base::start()
 	{
-		set_upload_only(true);
```

Comment:
The seed accounting is done based on `torrent_peer::seed`, as long as we only change `m_upload_only` it doesn't affect that. Eventually everything gets straitened out when the connection is established and `web_connection_base::on_connected` calls `incoming_have_all` which calls `set_seed`.

The call to `check_invariant` is way down the call stack:

`set_upload_only` -> `set_seed` -> `update_auto_sequential` -> `num_downloaders`